src/utils/staking.js

```diff
@@ -0,0 +1,278 @@
+import * as nearApiJs from 'near-api-js'
+import sha256 from 'js-sha256';
+import BN from 'bn.js'
+
+const {
+    transactions: {
+        functionCall
+    },
+    utils: {
+        format: {
+            parseNearAmount
+        }
+    }
+} =  nearApiJs
+
+// 125 - 175 Tgas for lockup methods
+const GAS_STAKE = '175000000000000'
+
+// custom lockups for testing (self deployed)
+const testLockup = {
+    'multisig.testnet': 'lu3.testnet',
+    'xa4.testnet': 'lu3.testnet',
+    'patrick12.testnet': 'patrick13.patrick12.testnet'
+}
+
+const stakingMethods = {
+	viewMethods: [
+		'get_account_staked_balance',
+		'get_account_unstaked_balance',
+		'get_account_total_balance',
+		'is_account_unstaked_balance_available',
+		'get_total_staked_balance',
+		'get_owner_id',
+		'get_reward_fee_fraction',
+	],
+	changeMethods: [
+		'ping',
+        'deposit',
+        'deposit_and_stake',
+        'deposit_to_staking_pool',
+		'stake',
+		'unstake',
+		'withdraw',
+	],
+}
+
+const lockupMethods = {
+    viewMethods: [
+        'get_balance',
+        'get_owners_balance',
+        'get_staking_pool_account_id',
+        'get_known_deposited_balance',
+    ]
+}
+
+const BETANET_WHITELIST = [
```

Comment:
why is this hardcoded? We need to load these from whitelist contract (specified when creating lockup)

https://explorer.testnet.near.org/transactions/7rqd2G1CQyMeU4kZoL7nUzdPiGydcAy6tGHfMvUGfBkF

e.g. in this transaction`"staking_pool_whitelist_account_id": "whitelist.f863973.m0"`

it should be possible to query from lockup AFAIK